novu/api/integration.py

```
@@ -109,3 +109,22 @@ def limit(self, channel: Channel) -> IntegrationChannelUsageDto:
         return IntegrationChannelUsageDto.from_camel_case(
             self.handle_request("GET", f"{self._integration_url}/{channel}/limit")["data"]
         )
+
+
+
+    def set_primary(self, integration_id: str) -> bool:
+        """Set an integration as primary
+
+        Args:
+            integration_id: The integration ID
+
+        Returns:
+            Is integration set active
+        """
+
+
+        try:
+            res = self.handle_request("POST", f"{self._integration_url}/{integration_id}/set-primary")["data"]
+            return res.get("primary") == True
+        except:
```

Broad-except aren't allowed, and i don't understand what you are trying to catch here ?

---

Now I have changed Broad-except to only accept the http errors. As when the endpoint set-primary is called with an invalid integration_id it responds with the 400 error. And in handle_request module of api/base.py the 400 error is not handled. Which is further leading to an crash of entire SDK and eventually entire application will crash.